modules/gapi/misc/python/pyopencv_gapi.hpp

```diff
@@ -295,6 +297,81 @@ PyObject* pyopencv_from(const GRunArgs& value)
     return list;
 }
 
+template<>
+PyObject* pyopencv_from(const GRunArgs& value)
+{
+    return pyopencv_from_run_args(value);
+}
+
+template<>
+PyObject* pyopencv_from(const GOptRunArgs& value)
+{
+    return pyopencv_from_run_args(value);
+}
+
+template <typename T>
+PyObject* pyopencv_from(const cv::optional<T>& opt)
+{
+    if (opt.has_value())
+    {
+        return pyopencv_from(*opt);
+    }
+    else
+    {
+        Py_RETURN_NONE;
+    }
+}
+
+template <>
+PyObject* pyopencv_from(const GOptRunArg& v)
+{
+    switch (v.index())
+    {
+        case GOptRunArg::index_of<cv::optional<cv::Mat>>():
+            return pyopencv_from(util::get<cv::optional<cv::Mat>>(v));
+
+        case GOptRunArg::index_of<cv::optional<cv::Scalar>>():
+            return pyopencv_from(util::get<cv::optional<cv::Scalar>>(v));
+
+        case GOptRunArg::index_of<optional<cv::detail::VectorRef>>():
+            return pyopencv_from(util::get<optional<cv::detail::VectorRef>>(v));
+
+        case GOptRunArg::index_of<optional<cv::detail::OpaqueRef>>():
+            return pyopencv_from(util::get<optional<cv::detail::OpaqueRef>>(v));
+    }
+
+    PyErr_SetString(PyExc_TypeError, "Failed to unpack GOptRunArg");
```

Comment:
I'd add here that index is unknown (as we don't have `default:` case here)

Reply:
Fixed.